src/transform/with_lengthscale.jl

```diff
@@ -0,0 +1,33 @@
+"""
+    with_lengthscale(kernel::Kernel, lengthscale::Real)
+    with_lengthscale(kernel::Kernel, lengthscales::AbstractVector{<:Real})
+
+Constructs a transformed kernel with `lengthscale`.
+If a vector `lengthscales` is passed instead, constructs an "ARD" kernel with different lengthscales for each dimension.
```

Comment:
The convention is to use the imperative form in the first line of the docstring:
```suggestion
Construct a transformed kernel with `lengthscale`.
```